code.gs

```diff
@@ -1389,20 +1406,29 @@ function main (forceDate) {
       .concat(isRunningOutdatedVersion() ? ['<br/><br/><b>', htmlEscape(bodySuffixes[2]), ' <a href="', baseGitHubProjectURL, 'releases/latest', '">', htmlEscape(bodySuffixes[3]), '</a>.</b></p>'] : ['</p>'])
       .join('');
 
-    // ...send the email notification.
-    log.add('Sending email...', 'info');
-    if (!settings.debug.log.logInstedOfMail) {
-      sendEmail(subject, body, htmlBody, inlineImages)
-    } else {
+    // ...log (if selected) and return mail content.
+    if (logEmail) {
       log.add('Subject: ' + subject);
       log.add('Content: ' + body);
       //log.add('Content HTML: ' + htmlBody); //is it needed?
     }
```

Comment:
I'd remove lines R1409-R1414 from this function because `generateEmailNotification` should only generate the email notification, whitout logging it: the notification should be logged only when running the unit tests.

By doing this we can have a clean flow without the use of arguments or settings for both the standard "run by the time trigger" case:

- `normal` is run by the trigger
- `normal` calls main
- `main` calls `generateEmailNotification` and stores the result
- `main` checks if there is any result
- `main` calls `sendEmail` passing the result of `generateEmailNotification`

and for the manual "unit test" case: 

- `unitTests` is run by the user
- `unitTests` calls `testSelectedPeriod`
- `testSelectedPeriod` calls `generateEmailNotification` and stores the result
- `testSelectedPeriod` logs the result (and possibly performs some kind of automated test on it)